File: implementations/java/hash_table/HashTable_large.java
package hash_table;

import java.io.*;

public class HashTable_Large {
    static final int N          = 100000;
    static final int TABLE_SIZE = 150001;

    static long[]    keys     = new long[TABLE_SIZE];
    static long[]    vals     = new long[TABLE_SIZE];
    static boolean[] occupied = new boolean[TABLE_SIZE];

    static void setup() {
        // reset handles initialization
    }

    static int hashFn(long key) {
        return (int)(((key * 2654435761L) & 0xFFFFFFFFFFFFFFFFL) % TABLE_SIZE);
    }

    static void insert(long key, long val) {
        int i = hashFn(key);
        while (occupied[i]) {
            if (keys[i] == key) { vals[i] = val; return; }
            i = (i + 1) % TABLE_SIZE;
        }
        keys[i]     = key;
        vals[i]     = val;
        occupied[i] = true;
    }

    static long lookup(long key) {
        int i = hashFn(key);
        while (occupied[i]) {
            if (keys[i] == key) return vals[i];
            i = (i + 1) % TABLE_SIZE;
        }
        return -1;
    }

    static void reset() {
        for (int i = 0; i < TABLE_SIZE; i++) occupied[i] = false;
        for (int i = 0; i < N; i++) insert((long)i * 7 + 3, (long)i * 13 + 5);
    }

    static long hashTable() {
        reset();
        long checksum = 0;
        for (int i = 0; i < N; i++) checksum += lookup((long)i * 7 + 3);
        return checksum;
    }

    public static void main(String[] args) throws IOException {
        setup();

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        System.out.println("ready");
        System.out.flush();

        String line;
        while ((line = br.readLine()) != null) {
            System.out.println(hashTable());
            System.out.flush();
        }
    }
}

File: implementations/javascript/hash_table/hash_table_large.js
const readline = require('readline');

const N          = 100000;
const TABLE_SIZE = 150001;

const keys     = new Array(TABLE_SIZE).fill(0);
const vals     = new Array(TABLE_SIZE).fill(0);
const occupied = new Array(TABLE_SIZE).fill(false);

function setup() {
    // reset handles initialization
}

function hash_fn(key) {
    return Number(BigInt(key) * 2654435761n & 0xFFFFFFFFFFFFFFFFn) % TABLE_SIZE;
}

function insert(key, val) {
    let i = hash_fn(key);
    while (occupied[i]) {
        if (keys[i] === key) { vals[i] = val; return; }
        i = (i + 1) % TABLE_SIZE;
    }
    keys[i]     = key;
    vals[i]     = val;
    occupied[i] = true;
}

function lookup(key) {
    let i = hash_fn(key);
    while (occupied[i]) {
        if (keys[i] === key) return vals[i];
        i = (i + 1) % TABLE_SIZE;
    }
    return -1;
}

function reset() {
    for (let i = 0; i < TABLE_SIZE; i++) occupied[i] = false;
    for (let i = 0; i < N; i++) insert(i * 7 + 3, i * 13 + 5);
}

function hash_table() {
    reset();
    let checksum = 0;
    for (let i = 0; i < N; i++) checksum += lookup(i * 7 + 3);
    return checksum;
}

setup();
process.stdout.write("ready\n");

const rl = readline.createInterface({ input: process.stdin });
rl.on('line', () => {
    process.stdout.write(String(hash_table()) + "\n");
});